tt_eager/tt_dnn/op_library/backward/backward_ops.cpp

```diff
@@ -119,21 +119,43 @@ std::vector<Tensor> exp_bw(const Tensor& grad, const Tensor& input, const Memory
 }
 
 
-std::vector<Tensor> _addcmul_bw(const Tensor& grad, const Tensor& input, const Tensor& tensor1, const Tensor& tensor2, float value, const MemoryConfig& output_mem_config) {
-    std::vector<Tensor> grad_tensor;
-    grad_tensor.emplace_back(grad);
-    Tensor grad_a = mul_unary(mul(grad, tensor2, std::nullopt, output_mem_config), value, output_mem_config);
-    grad_tensor.emplace_back(grad_a);
-    Tensor grad_b = mul_unary(mul(grad, tensor1, std::nullopt, output_mem_config), value, output_mem_config);
-    grad_tensor.emplace_back(grad_b);
+std::vector<std::optional<Tensor>> _addcmul_bw(const Tensor& grad, const Tensor& input, const Tensor& tensor1, const Tensor& tensor2, float value, const MemoryConfig& output_mem_config, const std::vector<bool> are_required_outputs, std::vector<std::optional<Tensor>> output_tensor) {
```

Comment:
Is it possible to return a `std::optional<Tensor>` and not return a std::vector of these?

Reply:
As backward ops send multiple output like grad, input_a and input_b we need the return type as vector.